src/ore/src/str.rs

```diff
@@ -436,4 +484,25 @@ mod tests {
         indent += 1;
         assert_eq!(indent.to_string(), "~~~".to_string());
     }
+
+    #[crate::test]
+    fn test_redact() {
+        // Don't pick up the soft-assert flag in this test, to emulate prod behaviour.
+        pub fn redact(s: impl Debug) -> impl Debug {
+            Redacting {
+                value: s,
+                debug_mode: false,
+            }
+        }
+
+        assert_eq!(
+            r#"<"XXXX_XXXXXX">"#,
```

Comment:
This felt like a plausible compromise - preserving some structure while redacting the majority of the content - but there are lots of other choices we could make here!